// @storybook/addon-themes@10.0.7/dist/index.js
import {
  preview_exports
} from "./_browser-chunks/chunk-R6XDDBXB.js";
import {
  DEFAULT_THEME_PARAMETERS,
  GLOBAL_KEY,
  PARAM_KEY,
  THEMING_EVENTS,
  __export,
  __name
} from "./_browser-chunks/chunk-D4FRO6NL.js";

// src/index.ts
import { definePreviewAddon } from "storybook/internal/csf";

// src/decorators/class-name.decorator.tsx
import { useEffect } from "storybook/preview-api";

// src/decorators/helpers.ts
var helpers_exports = {};
__export(helpers_exports, {
  initializeThemeState: () => initializeThemeState,
  pluckThemeFromContext: () => pluckThemeFromContext,
  useThemeParameters: () => useThemeParameters
});
import { deprecate } from "storybook/internal/client-logger";
import { addons, useParameter } from "storybook/preview-api";
import { dedent } from "ts-dedent";
function pluckThemeFromContext({ globals }) {
  return globals[GLOBAL_KEY] || "";
}
__name(pluckThemeFromContext, "pluckThemeFromContext");
function useThemeParameters(context) {
  deprecate(
    dedent`The useThemeParameters function is deprecated. Please access parameters via the context directly instead e.g.
    - const { themeOverride } = context.parameters.themes ?? {};
    `
  );
  if (!context) {
    return useParameter(PARAM_KEY, DEFAULT_THEME_PARAMETERS);
  }
  return context.parameters[PARAM_KEY] ?? DEFAULT_THEME_PARAMETERS;
}
__name(useThemeParameters, "useThemeParameters");
function initializeThemeState(themeNames, defaultTheme) {
  addons.getChannel().emit(THEMING_EVENTS.REGISTER_THEMES, {
    defaultTheme,
    themes: themeNames
  });
}
__name(initializeThemeState, "initializeThemeState");

// src/decorators/class-name.decorator.tsx
var DEFAULT_ELEMENT_SELECTOR = "html";
var classStringToArray = /* @__PURE__ */ __name((classString) => classString.split(" ").filter(Boolean), "classStringToArray");
var withThemeByClassName = /* @__PURE__ */ __name(({
  themes,
  defaultTheme,
  parentSelector = DEFAULT_ELEMENT_SELECTOR
}) => {
  initializeThemeState(Object.keys(themes), defaultTheme);
  return (storyFn, context) => {
    const { themeOverride } = context.parameters[PARAM_KEY] ?? {};
    const selected = pluckThemeFromContext(context);
    useEffect(() => {
      const selectedThemeName = themeOverride || selected || defaultTheme;
      const parentElement = document.querySelector(parentSelector);
      if (!parentElement) {
        return;
      }
      Object.entries(themes).filter(([themeName]) => themeName !== selectedThemeName).forEach(([themeName, className]) => {
        const classes = classStringToArray(className);
        if (classes.length > 0) {
          parentElement.classList.remove(...classes);
        }
      });
      const newThemeClasses = classStringToArray(themes[selectedThemeName]);
      if (newThemeClasses.length > 0) {
        parentElement.classList.add(...newThemeClasses);
      }
    }, [themeOverride, selected]);
    return storyFn();
  };
}, "withThemeByClassName");

// src/decorators/data-attribute.decorator.tsx
import { useEffect as useEffect2 } from "storybook/preview-api";
var DEFAULT_ELEMENT_SELECTOR2 = "html";
var DEFAULT_DATA_ATTRIBUTE = "data-theme";
var withThemeByDataAttribute = /* @__PURE__ */ __name(({
  themes,
  defaultTheme,
  parentSelector = DEFAULT_ELEMENT_SELECTOR2,
  attributeName = DEFAULT_DATA_ATTRIBUTE
}) => {
  initializeThemeState(Object.keys(themes), defaultTheme);
  return (storyFn, context) => {
    const { themeOverride } = context.parameters[PARAM_KEY] ?? {};
    const selected = pluckThemeFromContext(context);
    useEffect2(() => {
      const parentElement = document.querySelector(parentSelector);
      const themeKey = themeOverride || selected || defaultTheme;
      if (parentElement) {
        parentElement.setAttribute(attributeName, themes[themeKey]);
      }
    }, [themeOverride, selected]);
    return storyFn();
  };
}, "withThemeByDataAttribute");

// src/decorators/provider.decorator.tsx
import React from "react";
import { useMemo } from "storybook/preview-api";
var pluckThemeFromKeyPairTuple = /* @__PURE__ */ __name(([_, themeConfig]) => themeConfig, "pluckThemeFromKeyPairTuple");
var withThemeFromJSXProvider = /* @__PURE__ */ __name(({
  Provider,
  GlobalStyles,
  defaultTheme,
  themes = {}
}) => {
  const themeNames = Object.keys(themes);
  const initialTheme = defaultTheme || themeNames[0];
  initializeThemeState(themeNames, initialTheme);
  return (storyFn, context) => {
    const { themeOverride } = context.parameters[PARAM_KEY] ?? {};
    const selected = pluckThemeFromContext(context);
    const theme = useMemo(() => {
      const selectedThemeName = themeOverride || selected || initialTheme;
      const pairs = Object.entries(themes);
      return pairs.length === 1 ? pluckThemeFromKeyPairTuple(pairs[0]) : themes[selectedThemeName];
    }, [selected, themeOverride]);
    if (!Provider) {
      return React.createElement(React.Fragment, null, GlobalStyles && React.createElement(GlobalStyles, null), storyFn());
    }
    return React.createElement(Provider, { theme }, GlobalStyles && React.createElement(GlobalStyles, null), storyFn());
  };
}, "withThemeFromJSXProvider");

// src/index.ts
var index_default = /* @__PURE__ */ __name(() => definePreviewAddon(preview_exports), "default");
export {
  helpers_exports as DecoratorHelpers,
  index_default as default,
  withThemeByClassName,
  withThemeByDataAttribute,
  withThemeFromJSXProvider
};
