// @storybook/addon-themes@10.0.7/dist/postinstall.js
import CJS_COMPAT_NODE_URL_snyqki77rjp from 'node:url';
import CJS_COMPAT_NODE_PATH_snyqki77rjp from 'node:path';
import CJS_COMPAT_NODE_MODULE_snyqki77rjp from "node:module";

var __filename = CJS_COMPAT_NODE_URL_snyqki77rjp.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_snyqki77rjp.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_snyqki77rjp.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });

// src/postinstall.ts
import { spawn } from "child_process";
var PACKAGE_MANAGER_TO_COMMAND = {
  npm: "npx",
  pnpm: "pnpm dlx",
  yarn1: "npx",
  yarn2: "yarn dlx",
  bun: "bunx"
};
var selectPackageManagerCommand = /* @__PURE__ */ __name((packageManager) => PACKAGE_MANAGER_TO_COMMAND[packageManager], "selectPackageManagerCommand");
async function postinstall({ packageManager = "npm" }) {
  const command = selectPackageManagerCommand(packageManager);
  await spawn(`${command} @storybook/auto-config themes`, {
    stdio: "inherit",
    cwd: process.cwd(),
    shell: true
  });
}
__name(postinstall, "postinstall");
export {
  postinstall as default
};
